src/UI/Components/LogViewer/LogViewer.test.tsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { LogViewerComponent, LogViewerData } from "./LogViewer";
+
+// Mock data for testing
+const mockLogs: LogViewerData[] = [
+  {
+    id: "1",
+    name: "Log 1",
+    data: ["Line 1", "Line 2", "Line 3"],
+    duration: "1",
+    failed: false,
+  },
+  {
+    id: "2",
+    name: "Failed Log",
+    data: ["Error line 1", "Error line 2"],
+    duration: "20",
+    failed: true,
+  },
+];
+
+describe("LogViewerComponent", () => {
+  beforeEach(() => {
+    // Reset all mocks before each test
+    jest.resetAllMocks();
+  });
+
+  it("renders with initial log data", () => {
+    render(<LogViewerComponent logs={mockLogs} />);
+
+    // Check if first log name is displayed in select
+    expect(screen.getByRole("button", { name: /Log 1/i })).toBeInTheDocument();
+
+    // Check if log content is displayed
+    const logViewer = screen.getByTestId("log-viewer");
+    expect(logViewer).toHaveTextContent("Line 1");
+    expect(logViewer).toHaveTextContent("Line 2");
+    expect(logViewer).toHaveTextContent("Line 3");
+
+    // Check if duration is displayed
+    expect(screen.getByText(/Duration: 1 s/)).toBeInTheDocument();
+  });
+
+  it("switches between logs when selecting from dropdown", async () => {
+    render(<LogViewerComponent logs={mockLogs} />);
+
+    // Open dropdown
+    const toggle = screen.getByRole("button", { name: /Log 1/i });
+    await userEvent.click(toggle);
+
+    // Select second log
+    const option = screen.getByRole("option", { name: /Failed Log/i });
+    await userEvent.click(option);
+
+    // Check if new log content is displayed
+    const logViewer = screen.getByTestId("log-viewer");
+    expect(logViewer).toHaveTextContent("Error line 1");
+    expect(logViewer).toHaveTextContent("Error line 2");
+    expect(screen.getByText(/Duration: 200 ms/)).toBeInTheDocument();
```

Comment:
Duration is different value in the mock vs assertion, also here is 200 ms and above is 1s.